### PR TITLE
Refactor and fixes dropdown bugs

### LINKED ISSUE #8788: MenuItemWithOptionDropdowns are bugged
## Bug Description

https://github.com/user-attachments/assets/cde4d2c0-6b1a-4afe-ae29-8a356eb9949e

### LINKED ISSUE #8791: Sort Order Dropdown
Image

### LINKED ISSUE #8793: Regression - Margin top 2px
Image

### LINKED ISSUE #8890: Fix padding in multi item field input
Image

### LINKED ISSUE #8893: No scroll and search container is shrinking on Add people select on relation cards
Image

### PR DESCRIPTION
**Pull request by lucasbordeau**
Fixes https://github.com/twentyhq/twenty/issues/8788
Fixes https://github.com/twentyhq/twenty/issues/8793
Fixes https://github.com/twentyhq/twenty/issues/8791
Fixes https://github.com/twentyhq/twenty/issues/8890
Fixes https://github.com/twentyhq/twenty/issues/8893

- [x] Also : 

Icon buttons under dropdown are visible without blur : 

Capture d’écran du 2024-11-29 15-09-53

- [x] Also : 

image

- [x] Also : 

image

- [x] Also : 

image

- Button create new without padding
- Container is expanding

- [x] Also : 

image

- [x] Also : 

image

Font is cut at bottom ?

- [x] Also : 

image

The component should flip and not resize in this situation

- [x] Also : 

image

- [x] Also : 

image

### CODE REVIEW

**Comment by Bonapara** [comment]
related to https://github.com/twentyhq/twenty/issues/8200

File: packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCell.tsx [hunk 1]
```diff
@@ -37,6 +40,8 @@ export const RecordInlineCell = ({ loading }: RecordInlineCellProps) => {
 
   const { closeInlineCell } = useInlineCell();
 
+  const activeDropdownFocusId = useRecoilValue(activeDropdownFocusIdState);
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCell.tsx:43]: will this trigger a re-render of all cells when we click outside of an input

File: packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCell.tsx [hunk 2]
```diff
@@ -66,6 +71,16 @@ export const RecordInlineCell = ({ loading }: RecordInlineCellProps) => {
   };
 
   const handleClickOutside: FieldInputEvent = (persistField) => {
+    const recordFieldDropdownId = getDropdownFocusIdForRecordField(
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCell.tsx:74]: useRecoilCallback

File: packages/twenty-front/src/modules/spreadsheet-import/components/MatchColumnSelect.tsx [hunk 3]
```diff
@@ -101,6 +101,7 @@ export const MatchColumnSelect = ({
     callback: () => {
       setIsOpen(false);
     },
+    listenerId: 'MatchColumnSelect',
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/spreadsheet-import/components/MatchColumnSelect.tsx:104]: match-column-select

File: packages/twenty-front/src/modules/ui/field/input/components/AddressInput.tsx [hunk 4]
```diff
@@ -190,18 +191,22 @@ export const AddressInput = ({
...
+      }
+
       event.stopImmediatePropagation();
 
       closeCountryDropdown();
-
       onClickOutside?.(event, internalValue);
     },
     enabled: isDefined(onClickOutside),
+    listenerId: 'addressInput',
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/field/input/components/AddressInput.tsx:209]: address-input

File: packages/twenty-front/src/modules/ui/field/input/components/DoubleTextInput.tsx [hunk 5]
```diff
@@ -158,6 +158,7 @@ export const DoubleTextInput = ({
       });
     },
     enabled: isDefined(onClickOutside),
+    listenerId: 'DoubleTextInput',
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/field/input/components/DoubleTextInput.tsx:161]: double-text-input

File: packages/twenty-front/src/modules/ui/input/components/SelectInput.tsx [hunk 6]
```diff
@@ -113,6 +77,7 @@ export const SelectInput = ({
         onCancel();
       }
     },
+    listenerId: 'SelectInput',
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/input/components/SelectInput.tsx:80]: select-input

File: packages/twenty-front/src/modules/ui/input/components/internal/currency/components/CurrencyPickerDropdownSelect.tsx [hunk 7]
```diff
@@ -32,7 +32,7 @@ export const CurrencyPickerDropdownSelect = ({
   );
 
   return (
-    <DropdownMenu width="240px" disableBlur>
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/input/components/internal/currency/components/CurrencyPickerDropdownSelect.tsx:35]: don't we need this 240px + double check currency picker

File: packages/twenty-front/src/modules/ui/layout/dropdown/components/Dropdown.tsx [hunk 8]
```diff
@@ -51,7 +44,6 @@ type DropdownProps = {
 };
 
 export const Dropdown = ({
-  className,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/dropdown/components/Dropdown.tsx:54]: leave

File: packages/twenty-front/src/modules/ui/layout/dropdown/components/DropdownMenu.tsx [hunk 9]
```diff
@@ -23,7 +23,7 @@ const StyledDropdownMenu = styled.div<{
   display: flex;
 
   flex-direction: column;
-  z-index: 30;
+  z-index: ${({ theme }) => theme.zIndex.dropdown};
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/dropdown/components/DropdownMenu.tsx:26]: zIndex is not an absolute concept. It cannot be part of the theme

File: packages/twenty-front/src/modules/ui/layout/dropdown/utils/getDropdownFocusIdForRecordField.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+export const getDropdownFocusIdForRecordField = (
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/dropdown/utils/getDropdownFocusIdForRecordField.ts:1]: this should be part of object-record

File: packages/twenty-front/src/modules/ui/layout/expandable-list/components/ExpandedListDropdown.tsx [hunk 11]
```diff
@@ -46,8 +46,11 @@ export const ExpandedListDropdown = ({
   });
 
   useListenClickOutside({
-    refs: [refs.floating],
-    callback: onClickOutside ?? (() => {}),
+    refs: [refs.domReference],
+    callback: () => {
+      onClickOutside?.();
+    },
+    listenerId: 'ExpandableList',
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/expandable-list/components/ExpandedListDropdown.tsx:53]: expandable-list

File: packages/twenty-front/src/modules/ui/layout/modal/components/Modal.tsx [hunk 12]
```diff
@@ -30,7 +30,7 @@ const StyledModalDiv = styled(motion.div)<{
     return theme.border.radius.md;
   }};
   overflow: hidden;
-  z-index: 10000; // should be higher than Backdrop's z-index
+  z-index: ${({ theme }) => theme.zIndex.modal} // should be higher than Backdrop's z-index
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/modal/components/Modal.tsx:33]: same

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx [hunk 13]
```diff
@@ -40,7 +33,7 @@ const StyledContainer = styled(motion.div)<{ isRightDrawerMinimized: boolean }>`
 
   right: 0;
   top: 0;
-  z-index: 100;
+  z-index: ${({ theme }) => theme.zIndex.rightDrawer};
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx:36]: same

**Review by charlesBochet — APPROVED** [review]
LGTM

**Pull request merged**